keps/sig-storage/2848-transfer-volumesnapshots-namespaces/README.md

```diff
@@ -0,0 +1,795 @@
+  - [ ] (R) Ensure GA e2e tests for meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+This KEP proposes a method for transferring VolumeSnapshot between namespaces. 
+
+Currently, VolumeSnapshot, and VolumeSnapshotContent are bound to specific namespace. The binding relationship between these two storage resources is one-to-one.
+This proposal targets allowing users to conveniently move these resources without manual creation
+of each independent object (VolumeSnapshot, VolumeSnapshotContent).
+
+Based on community design sessions, and inspired by [API for PVC Namespace Transfer](https://github.com/kubernetes/enhancements/pull/2110/files), the following is being proposed :-
+
+- `VolumeSnapshotTransferRequest` resources are created in the target namespace.
```

Comment:
Is my understanding right that there are changes in the model who create `VolumeSnapshotTransferRequest` and `VolumeSnapshotTransferApproval` in which namespace?

Current model:
1. `VolumeSnapshotTransferRequest` is created in `target-namespace` where the snapshot should be transferred by a user who doesn't own the snapshot currently.
2. `VolumeSnapshotTransferApproval` is created in `source-namespace` where the snapshot currently exists by an owner of the snapshot.

Previous model:
1. `StorageTransferRequest` is created in `source-namespace` where the snapshot currently exists by an owner of the snapshot.
2. `StorageTransferAccept` is created in `target-namespace` where the snapshot should be transferred by a user who doesn't own the snapshot currently.

I prefer the current model, because these changes with other modifications will provide a guarantee:
- 1:1 mapping of `VolumeSnapshotTransferRequest` and `VolumeSnapshotTransferApproval` (No other approvals in other namespaces)
- 1:1 mapping of `VolumeSnapshotTransferRequest` and target snapshot (No other snapshots in other namespaces)

We still need to handle other corner cases like multiple `VolumeSnapshotTransferRequest`s for one source snapshot, but it seems a big improvement.

Reply:
I slightly prefer `VolumeSnapshotTransferRequest` is created in the source namespace too. It could be easier to control that only one transfer of a snapshot happens at a time. For example by requiring `VolumeSnapshot.Name == VolumeSnapshotTransferRequest.Name`